### PR TITLE
feat: simplify args inputs for `terraform_docs`

### LINKED ISSUE #154: wrong arguments passed to terraform-docs (v0.10.0 and v0.10.1
Related: https://github.com/terraform-docs/terraform-docs/issues/324

```

### PR DESCRIPTION
**Pull request by 1oglop1**
This should fix #154.
I removed `getopt` and instead simply passed args array in the same order as specified via `.pre-commit-config.yaml` file.

example usage:

```
  - repo: https://github.com/antonbabenko/pre-commit-terraform
    rev: <version>
    hooks:
      - id: terraform_docs
        verbose: true
        name: Terraform docs
        description: Inserts input and output documentation into README.md (using terraform-docs).
        require_serial: true
        entry: _ignore/pre-commit-terraform/terraform_docs.sh
        language: script
        files: (\.tf)$
        exclude: \.terraform\/.*$
        args:
          - --header-from=doc.tf
          - --hide-all
          - md
```

### CODE REVIEW

**Comment by 1oglop1** [comment]
@antonbabenko Take this as an initial proposal, there is some additional cleanup required but I'll put more effort after discussion.

**Comment by 1oglop1** [comment]
nudge @antonbabenko could you please provide your opinion on this? PR received a few reactions from which I assume it's a wanted feature by more people.

Thank you.

**Comment by antonbabenko** [comment]
Hi @1oglop1 ! There is simply not enough time for me to review all the projects I am involved in. I assigned @robinbowes to take a look if he has some time.

**Comment by GenPage** [comment]
@robinbowes Any update on your review?

**Comment by 1oglop1** [comment]
@Vlaaaaaaad I noticed you liked my PR, and since @antonbabenko takes preference on issues/PRs from sponsors it may help if you participate in the discussion to give @antonbabenko incentive to look and comment on this change

**Comment by maulik-vimond** [comment]
Can this be merged? Out of the box terraform-docs does not work right now unless we downgrade terraform-docs.

**Comment by robinbowes** [comment]
> does not work right now

What doesn't work? It's working for me.

```
repos:
- repo: https://github.com/pre-commit/pre-commit-hooks
  rev: v3.4.0
  hooks:
    - id: check-yaml
    - id: trailing-whitespace
    - id: end-of-file-fixer
- repo: https://github.com/antonbabenko/pre-commit-terraform
  rev: v1.47.0
  hooks:
    - id: terraform_validate
      args:
        - '--envs=AWS_DEFAULT_REGION=us-east-1'
    - id: terraform_fmt
    - id: terraform_docs
```

```
 $ terraform-docs --version
terraform-docs version v0.11.2 darwin/amd64 BuildDate: 2021-02-23T00:25:32+0000
```

**Comment by maulik-vimond** [comment]
> What doesn't work? It's working for me.

How is your terraform code structured? Mine is in a sub-folder. Please see the issue #181.

File: terraform_docs.sh [hunk 1]
```diff
@@ -125,7 +126,7 @@ terraform_docs() {
 
     if [[ "$terraform_docs_awk_file" == "0" ]]; then
       # shellcheck disable=SC2086
-      terraform-docs md $args ./ > "$tmp_file"
+      terraform-docs $args ./ > "$tmp_file"
```
Comment (maulik-vimond) [inline terraform_docs.sh:129]: To provide backward compatibility shouldn't $args have md default somewhere?

**Comment by ezio-ballarin** [comment]
I also noticed I can't pass args to the terraform-docs hook. here's my config: 

```
repos:
- repo: git://github.com/antonbabenko/pre-commit-terraform
  rev: v1.50.0 
  hooks:
    - id: terraform_fmt
    - id: terraform_docs
      args: 
        - '--args=--sort-by=required --hide=providers'
```
```
~$ pre-commit -V
pre-commit 2.13.0
~$ terraform-docs version
terraform-docs version v0.13.0 linux/amd64 BuildDate: 2021-05-27T15:35:35-07:00
```

File: terraform_docs.sh [hunk 2]
```diff
@@ -20,30 +20,29 @@ initialize_() {
...
-      --)
-        shift
-        FILES=("$@")
-        break
-        ;;
-    esac
+  argv=($@)
+
+  for arg in ${argv[@]}; do
+    echo WORKING with: \"$arg\"
```
Comment (7adityaraj) [inline terraform_docs.sh:30]: We should also hide this print or should echo when verbose is enabled.

**Comment by MaxymVlasov** [comment]
Too outdated

**Pull request closed without merging**